manual/src/ornate/reference.md

```diff
@@ -64,24 +64,94 @@ useYarn := true
+file that can weigh several MBs if your application is large. Scalajs-bundler provides
+a few different options with respect to handling this large output file.
 
-You can get a faster “change source and reload application” workflow by setting the `enableReloadWorkflow` 
-key to `true`. An alternative way to invoke reload workflow is using the `fastOptJs::webpackReload` task.
+#### Application (default) {#bundling-mode-application}
 
-The reload workflow replaces the `fastOptJS::webpack` task implementation with a different one, that does not use 
-webpack to process the output of the Scala.js compilation.  Instead, it pre-bundles the modules imported by your 
-application and exposes them to the global namespace. Since these dependencies can then be resolved from the global 
-namespace, the output of Scala.js is just concatenated after the contents of the pre-bundling process.
+`webpackBundlingMode := BundlingMode.Application` generates a webpack config that simply processes the scala.js output file as an 
+entrypoint. This means that webpack loaders, plugins, etc will work as usual. It also means that webpack will 
+have to process a very large Scala.Js output file.
 
-It is possible to configure an alternative webpack configuration file which is used for building the bundle using the
-`webpackConfigFile in webpackReload` setting . The configuration file may not contain `entry` nor `output` configuration
-but can be used to for loaders etc.
+Turning this huge CommonJS module into code executable by web browsers takes time, often upwards of a minute. This
+problem is even worse when processing (larger) fastopt files, which eliminates the point of generating
+them.
+
+Nonetheless, this is identical to what you'd get if were to duplicate your workflow outside scalajs-bundler, 
+so it remains the default. 
+
+#### Library Only {#bundling-mode-library-only}
+You can get much a faster “change source and reload application” workflow by setting the 
+`webpackBundlingMode := BundlingMode.LibraryOnly()` key. This bundling mode avoids having webpack process the entire
+scala.js output, but instead uses webpack to build all the javascript dependencies (determined via `@JSImport` 
+and any changes to the `webpack.config.js`).
+
+It then uses a special "loader" to resolve these dependencies for the actual scala.js application.
+
+In order to use this mode, instead of including `yourapp-bundle.js` in your page, you will need to include
+`yourapp-library.js`, `yourapp-loader.js`, `yourapp-fastopt.js` or `yourapp-opt.js`. If you're using Play 
+Framework, you could use a twirl template similar to this one:
+
+~~~ scala
+@(projectName: String,
+  assets: String => String,
+  resourceExists: String => Boolean,
+  htmlAttributes: Html = Html("")
+)
+
+@defining(s"${projectName.toLowerCase}") { name =>
+  @Seq(s"$name-opt-library.js", s"$name-fastopt-library.js").find(resourceExists).map(name => jsScript(assets(name), htmlAttributes))
```

Comment:
What’s `jsScript`? Would it be possible to include that into the `play` test and build the documentation from that test?

Reply:
JsScript comes from https://github.com/vmunier/scalajs-scripts. I think the correct solution here is to either push this script upstream or pull scalajs-scripts into scalajs-bundler as a scalajs-bundler-play dependency or something. I dropped the script in the docs as a dummy example, but if it's more confusing than not, we can just drop it entirely.